### ISSUE TITLE
Chinese name of test case can not display correct on console log

### ISSUE THREAD

**Issue by mark26195491** [robotframework/RIDE#2361]
RIDE v2.0b2.dev1 running on Python 3.8.5. 

Package                   Version
------------------------- -----------
bcrypt                    3.2.0
cffi                      1.14.4
cryptography              3.3.1
numpy                     1.19.5
paramiko                  2.7.2
Pillow                    8.1.0
pip                       21.0
psutil                    5.8.0
pycparser                 2.20
Pygments                  2.7.4
PyMySQL                   1.0.2
PyNaCl                    1.4.0
Pypubsub                  4.0.3
pywin32                   300
robotframework            3.2.2
robotframework-ride       2.0b2.dev1
robotframework-sshlibrary 3.6.0
scp                       0.13.3
setuptools                47.1.0
six                       1.15.0
wxPython                  4.0.7.post2

![image](https://user-images.githubusercontent.com/55422618/105944246-e8fda100-609d-11eb-8b02-87774de56b07.png)

**Comment by mark26195491** [comment]
Exception:
![image](https://user-images.githubusercontent.com/55422618/105954935-6c27f280-60b0-11eb-8036-f4fe40be6c3f.png)

**Comment by HelioGuilherme66** [comment]
Please try with current development version.
Then, try to upgrade wxPython.

**Comment by mark26195491** [comment]
> Please try with current development version.
> Then, try to upgrade wxPython.

Currently， i have already use the dev version      RIDE v2.0b2.dev1

**Comment by HelioGuilherme66** [comment]
Then, try to upgrade wxPython.

Please paste output of command:
`chcp`

**Comment by panational** [comment]
@HelioGuilherme66 ：I have the same problem：(it not display correct on console log but it's  display correct on Message log   )
![image](https://user-images.githubusercontent.com/24242088/106406702-16b76100-6475-11eb-81a4-093410576b5b.png)
The environment is as follows：
![image](https://user-images.githubusercontent.com/24242088/106406528-7c571d80-6474-11eb-96a6-a894c0093ad1.png)
and
executes the chcp command：
活动代码页: 936

**Comment by clclclclclclcl** [comment]
![image](https://user-images.githubusercontent.com/49467409/115823236-8e73a700-a438-11eb-8bd4-cd7fc9508f8c.png)

**Comment by yuwy9988** [comment]
python3.9 543行不是这个

**Comment by yuwy9988** [comment]
![image](https://user-images.githubusercontent.com/89013845/129555367-060d80e5-47ae-4313-b683-bf232e8dfaa1.png)

**Comment by HelioGuilherme66** [comment]
Need the output of command `chcp` in a command window.
This is mine:
```
C:\>chcp
Active code page: 850
```

And you should try running `robot` in a command window to confirm the console log shows OK.

This is only a visual problem on RIDE, for the Console Log, not for the Output Log.

**Comment by yuwy9988** [comment]
change post request ok

**Comment by yuwy9988** [comment]
How to set when the label page disappears run tag

**Comment by HelioGuilherme66** [comment]
@mark26195491 Please retest on new version 2.0b2.

**Comment by zhujunling-nj** [comment]
RIDE v2.0b2
wxPython  4.1.1
Python 3.8.10
Windows 10

Print the error for Chinese name of test case:
Traceback (most recent call last):
  File "C:\Python38\lib\site-packages\robotide\contrib\testrunner\testrunnerplugin.py", line 575, in OnTimer
    self._append_to_console_log(out_buffer, source="stdout")
  File "C:\Python38\lib\site-packages\robotide\contrib\testrunner\testrunnerplugin.py", line 605, in _append_to_console_log
    self._append_text(self._console_log_ctrl, text, source)
  File "C:\Python38\lib\site-packages\robotide\contrib\testrunner\testrunnerplugin.py", line 626, in _append_text
    text_ctrl.AppendText(text)
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xb4 in position 12: invalid start byte

I modify the  site-packages\robotide\contrib\testrunner\testrunnerplugin.py, and then fix the error.
![2022-10-09_141306](https://user-images.githubusercontent.com/26447863/194741082-a8510b11-e3ee-421a-8b61-3ced4481bca3.png)

**Comment by HelioGuilherme66** [comment]
Please see my comment on #2494

**Comment by HelioGuilherme66** [comment]
This is another test case with similar problems, even applying the #2494 will not show  the expected output.
```
*** Test Cases ***
TC1
    [Tags]    console
    Log to console    \nHello world!
    Log to console    Diacritics: éëè
    Log to console    Hello world again!
TC2
    [Tags]    console
    Log to console    \nHello world!
    Log to console    Hello world again!
```

**Comment by zhujunling-nj** [comment]
> This is another test case with similar problems, even applying the #2494 will not show the expected output.
> 
> ```
> *** Test Cases ***
> TC1
>     [Tags]    console
>     Log to console    \nHello world!
>     Log to console    Diacritics: éëè
>     Log to console    Hello world again!
> TC2
>     [Tags]    console
>     Log to console    \nHello world!
>     Log to console    Hello world again!
> ```
I set windows non-unicode to Dutch and modified encoding ['CONSOLE'] in testrunnerplugin.py to encoding ['SYSTEM'],  the test case TC1 seems to be working properly.

![2022-11-03_090116](https://user-images.githubusercontent.com/26447863/199635288-5794c1f8-8118-4879-86d1-37b12697f621.png)

**Comment by zhujunling-nj** [comment]
In the Dutch language Windows, OEM Code Page is 850, ANSI Code Page is 1252.
The OEMCP apply to the console system calls, and the ANSICP apply to the file system calls. Read and write the robot's STDOUT seem to be file operation.

**Comment by Harm10** [comment]
@zhujunling-nj Your suggestion for encoding SYSTEM also works with me (in essence this is using locale.getpreferredencoding()) but the Chinese test case does not work properly. The Chinese characters all end up as ? characters. Is this also the case when you run it with that SYSTEM encoding?

**Comment by zhujunling-nj** [comment]
> @zhujunling-nj Your suggestion for encoding SYSTEM also works with me (in essence this is using locale.getpreferredencoding()) but the Chinese test case does not work properly. The Chinese characters all end up as ? characters. Is this also the case when you run it with that SYSTEM encoding?

When I set Winodws to Chinese (code page 936), the Chinese test case runs normal, but TC1 runs abnormally.
In winodws systems, you cannot support two or more non-unicode charsets simultaneously.

**Comment by zhujunling-nj** [comment]
In python 3, I use another solution (without the [2494](https://github.com/robotframework/RIDE/pull/2494)), it seems to be working properly.
Set the windows environment PYTHONIOENCODING to UTF-8, and no source code need to be modified. And this solution supports multi-languages.
If this solution also applies to someone else, I will close the [2494](https://github.com/robotframework/RIDE/pull/2494)

**Comment by Harm10** [comment]
What is the best way to set PYTHONIOENCODING to UTF-8? Should it only be set for RIDE execution or RF in general? The latter should also have effect on command line execution?

**Comment by zhujunling-nj** [comment]
I have tried two solutions:
1. Set the RIDE shortcut target to  _C:\Python38\pythonw.exe -c "import os; from robotide import main; os.environ['PYTHONIOENCODING'] = 'UTF-8'; main()"_. This is only effective for the RIDE and RF runned by RIDE.
2. Set the system environment PYTHONIOENCODING to UTF-8.
The latter may cause an abnormal display of the python programs running on some platforms (such as Jenkins, etc.). These platforms parse the output of the program with a ansi encoding.

**Comment by zhujunling-nj** [comment]
> I have tried two solutions:
> 
> 1. Set the RIDE shortcut target to  _C:\Python38\pythonw.exe -c "import os; from robotide import main; os.environ['PYTHONIOENCODING'] = 'UTF-8'; main()"_. This is only effective for the RIDE and RF runned by RIDE.
> 2. Set the system environment PYTHONIOENCODING to UTF-8. This effective for all python programs.
> 
> The latter may cause an abnormal display of the python programs running on some platforms (such as Jenkins, etc.). These platforms parse the output of the program with a ansi encoding.

Another solution: Modity robotide\contrib\testrunner\Process.py，add one line as follows.
![image](https://user-images.githubusercontent.com/26447863/200489851-a22fab80-0ab7-44f1-822f-020ca44b84d1.png)
This is only effective for the  RF runned by RIDE on windows.

**Comment by Harm10** [comment]
@HelioGuilherme66 I quite like the solution in Process.py. You are then totally independent of what is set outside RIDE. Do you want me to test this?

@zhujunling-nj Maybe you have seen my own PR for testrunnerplugin.py? As this looks like a Windows problem only I tried to make the fix more specific. Maybe you could also have a look?

**Comment by HelioGuilherme66** [comment]
I will test @zhujunling-nj PR, because is already updated to be Windows only, and seems a good solution. Only today I could reproduce the problem. This has been a problem since around 2009. We fix in RIDE, but then some changes in RF brings it back.

**Comment by HelioGuilherme66** [comment]
Closed by #2494

**Comment by zhujunling-nj** [comment]
> @zhujunling-nj Maybe you have seen my own PR for testrunnerplugin.py? As this looks like a Windows problem only I tried to make the fix more specific. Maybe you could also have a look?

Yes, I looked at the PR [#2502](https://github.com/robotframework/RIDE/pull/2502).
